### PR TITLE
Add namespace state strategy

### PR DESCRIPTION
**Pull request by yaron2**
Closes https://github.com/dapr/dapr/issues/4676.

Docs PR: https://github.com/dapr/docs/pull/2474.

### CODE REVIEW

File: pkg/components/state/state_config.go [hunk 1]
```diff
@@ -68,6 +73,15 @@ func GetModifiedStateKey(key, storeName, appID string) (string, error) {
 			return key, nil
 		}
 		return fmt.Sprintf("%s%s%s", appID, daprSeparator, key), nil
+	case strategyNamespace:
+		if appID == "" {
+			return key, nil
```
Comment (daixiang0) [inline pkg/components/state/state_config.go:78]: Please add this case to tests.

Comment (yaron2) [reply pkg/components/state/state_config.go:78]: Added.

**Comment by daixiang0** [comment]
@yaron2 now can we ignore fossa CI? seems always fail.

**Comment by yaron2** [comment]
> @yaron2 now can we ignore fossa CI? seems always fail.

Yeah, its getting fixed.

**Review by daixiang0 — APPROVED** [review]
LGTM, wait for CI happy.

**Comment by yaron2** [comment]
Thanks for the review @daixiang0

**Pull request merged**